### ISSUE TITLE
Dynamic Variables with Newman

### ISSUE THREAD

**Issue by krisgutta** [postmanlabs/newman#377]
It looks like Dynamic Variables are not supported in newman? If so, do you have plans for adding support?

I am using {{$timestamp}}, {{$randomInt}} and works just fine from Postman, but newman doesn't substitute with the actual values when executed in the command line. I am running node v4.4.2 and newman v2.0.5

Thanks,
-Kris

**Comment by czardoz** [comment]
Hi Kris,

I wasn't able to reproduce this issue. I created a collection: https://www.getpostman.com/collections/ae4651f73d3b232a43a9

Where the request body contains `{{$timestamp}}`and `{{$randomInt}}`, and the sent request contained the substituted values:
```
{
    "timestamp": "1459847095087",
    "randomInt": "397"
}
```
The substitution is working with headers too. 

Could you perhaps share a collection where this is not working for you?

**Comment by krisgutta** [comment]
Hi,

Thanks for the quick response, I've done some testing and it looks like the issue is with POST and not GET. Here is my newman command, newman -u https://www.getpostman.com/collections/ebdf54e05f85cf0c4fc5 -O Downloads/dynamic_bug.json -e Downloads/Test.json

You can get the Test.json file from https://drive.google.com/a/twilio.com/folderview?id=0B5JXE0oJeowgODg2NjRPN1FvSTQ&usp=sharing

I also included dynamic_bug.json file with request/response data in the same folder.

If you have trouble accessing my google drive, please note that I am not using anything from Test.json in the collection.

Thanks.

**Comment by czardoz** [comment]
Fixed this on the develop branch, and it'll be out in the next release (sometime this week) :cake:

**Comment by krisgutta** [comment]
Awesome ... Thanks very much, i will keep a look out for the release.

**Comment by krisgutta** [comment]
Is there an ETA on this update?

**Comment by czardoz** [comment]
Hi, this was released in `v2.0.7`. Updating Newman should bring in the fix.

**Comment by ok11** [comment]
Seems to not work in 2.0.9
What I'm doing, is setting a random user name in a prerequest script:
`"preRequestScript": "postman.setEnvironmentVariable(\"userLogin\", \"user{{$randomInt}}\");\n",
`
and then use in a request body
```
{
...
"userLogin": "{{userLogin}}",
...
}
```
Works fine in Postman, what I get as a `userLogin` in Newman, is `user{{$randomInt}}`.

I must admit, I'm new at Newman, maybe I misunderstand the concept, but I thought it should work exactly the same way as in Postman...

**Comment by willybill** [comment]
I updated newman on my windows workstation to 2.1.0 and am running 2.0.9 on the linux server and I am still seeing this behavior in both places.  In my data file I set variable "filename": "RunnerTestDoc{{$timestamp}}" and when I run a test using Postman Runner it sets filename to (RunnerTestDoc1464285327.docx) but when I run the same test using newman it sets filename to (RunnerTestDoc{{$timestamp}}.docx).  Any suggestions?
